Documentation updates for clarifying what is allowed in LIMIT & Co

arnefischereit opened the pull request.

arnefischereit committed:
Documentation updates for clarifying what is allowed in LIMIT/SKIP/OF ... ROWS

neo-technology-commit-status-publisher commented:
Looks like you've updated the documentation!

Check out your changes at https://neo4j-docs-cypher-588.surge.sh
<!-- Sticky Pull Request Comment -->

AzuObs commented on modules/ROOT/pages/clauses/call-subquery.adoc:
```diff
@@ -534,7 +538,9 @@ The amount of work to do in each separate transaction can be specified in terms
-The following is the same example but with one transaction every `2` input rows:
+The number of rows can be expressed using any expression that evaluates to a positive integer -- however the expression cannot refer to nodes or relationships.
+
+The following is the same example as above for loading a CSV file but with one transaction every `2` input rows:
```
I am not sure whether we should backquote `2` (and also`1000`) a couple lines above. But I leave it up to you or the documentation team to decide on style.

JPryce-Aklundh replied:
Both should be within backticks (`1000` because it is referring to a default value, and `2` because it references something in a code example).

AzuObs reviewed (approved):
Nice job. The comment is optional 😄

JPryce-Aklundh commented on modules/ROOT/pages/clauses/call-subquery.adoc:
```diff
@@ -534,7 +538,9 @@ The amount of work to do in each separate transaction can be specified in terms
 The number of input rows is set with the modifier `OF n ROWS` (or `ROW`).
 If omitted, the default batch size is `1000` rows.
-The following is the same example but with one transaction every `2` input rows:
+The number of rows can be expressed using any expression that evaluates to a positive integer -- however the expression cannot refer to nodes or relationships.
```
```suggestion
The number of rows can be expressed using any expression that evaluates to a positive integer and does not refer to nodes or relationships.
```

JPryce-Aklundh commented on modules/ROOT/pages/clauses/call-subquery.adoc:
```diff
@@ -534,7 +538,9 @@ The amount of work to do in each separate transaction can be specified in terms
-The following is the same example but with one transaction every `2` input rows:
+The number of rows can be expressed using any expression that evaluates to a positive integer -- however the expression cannot refer to nodes or relationships.
+
+The following is the same example as above for loading a CSV file but with one transaction every `2` input rows:
```
```suggestion
This example loads a CSV file with one transaction for every `2` input rows:
```

arnefischereit replied:
So, you do not deem the information important, that this is only a slight modification of the previous example?

JPryce-Aklundh replied:
Not really, no. I think it works as a stand-alone example, without tempting the reader to scroll back up again.

JPryce-Aklundh commented on modules/ROOT/pages/clauses/limit.adoc:
```diff
@@ -59,7 +59,7 @@ d|Rows: 3
 == Using an expression with `LIMIT` to return a subset of the rows
 
-Limit accepts any expression that evaluates to a positive integer as long as it is not referring to any external variables:
+`LIMIT` accepts any expression that evaluates to a positive integer, as long as it can be statically calculated:
```
"statically calculated" should probably be briefly defined - what do you think about either of the below 2 suggestions?

```suggestion
`LIMIT` accepts any expression that evaluates to a positive integer, as long as it can be calculated before the query is run."
```

Alternatively: "...as long as it can be statically calculated (i.e. calculated before the query is run)."

JPryce-Aklundh commented on modules/ROOT/pages/clauses/skip.adoc:
```diff
@@ -87,7 +87,7 @@ d|Rows: 2
 == Using an expression with `SKIP` to return a subset of the rows
 
-Skip accepts any expression that evaluates to a positive integer as long as it is not referring to any external variables:
+`SKIP` accepts any expression that evaluates to a positive integer, as long as it can be statically calculated:
```
See above comment.

arnefischereit committed:
Code Review suggestions

arnefischereit committed:
Clarification

arnefischereit committed:
Code review suggestion

Simplify description of LOAD CSV example: Do not mention previous similar example.

Co-authored-by: Jens Pryce-Åklundh <112686610+JPryce-Aklundh@users.noreply.github.com>

JPryce-Aklundh committed:
Merge branch 'dev' into skip-limit-limitations
